The job **`x86_64-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/19544313205/job/55959162783) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/55959162783) [(plain)](https://github.com/rust-lang/rust/commit/06596c0e17548161562b26b65bf246eb4fcd378a/checks/55959162783/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
---- [ui] tests/ui/sanitizer/dataflow.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/sanitizer/dataflow.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitizer/dataflow/a" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-Zsanitizer=dataflow" "-Zsanitizer-dataflow-abilist=/checkout/tests/ui/sanitizer/dataflow-abilist.txt" "-C" "unsafe-allow-abi-mismatch=sanitizer"
stdout: none
--- stderr -------------------------------
error: linking with `cc` failed: exit status: 1
   |
   = note:  "cc" "-m64" "/tmp/rustcX99GJO/symbols.o" "-Wl,-Bstatic" "-Wl,--whole-archive" "-Wl,<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc-nightly_rt.dfsan.a" "-Wl,--no-whole-archive" "<1 object files omitted>" "-Wl,--as-needed" "-Wl,-Bdynamic" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-231135333275c1cf.so" "-Wl,-Bstatic" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-*.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-L" "/tmp/rustcX99GJO/raw-dylibs" "-B<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld" "-B<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld" "-fuse-ld=lld" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitizer/dataflow/a" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-Wl,-O1" "-Wl,--strip-debug" "-nodefaultlibs" "-Wl,-rpath,$ORIGIN/../../../../stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib,--enable-new-dtags,-z,origin"
   = note: some arguments are omitted. use `--verbose` to show all linker arguments
   = note: rust-lld: error: undefined symbol: core::panicking::assert_failed_inner (.dfsan)
           >>> referenced by dataflow.a934cfd234e75439-cgu.0
           >>>               /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitizer/dataflow/a.dataflow.a934cfd234e75439-cgu.0.rcgu.o:(core::panicking::assert_failed::<u8, u8> (.dfsan))
           
           rust-lld: error: undefined symbol: core::panicking::assert_failed::<i32, i32> (.dfsan)
           >>> referenced by dataflow.a934cfd234e75439-cgu.0
           >>>               /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitizer/dataflow/a.dataflow.a934cfd234e75439-cgu.0.rcgu.o:(dataflow::main (.dfsan))
           >>> referenced by dataflow.a934cfd234e75439-cgu.0
           >>>               /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitizer/dataflow/a.dataflow.a934cfd234e75439-cgu.0.rcgu.o:(dataflow::main (.dfsan))
           >>> referenced by dataflow.a934cfd234e75439-cgu.0
           >>>               /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitizer/dataflow/a.dataflow.a934cfd234e75439-cgu.0.rcgu.o:(dataflow::main (.dfsan))
           >>> referenced 1 more times
           
           rust-lld: error: undefined symbol: <u8 as core::fmt::Display>::fmt (.dfsan)
           >>> referenced by dataflow.a934cfd234e75439-cgu.0
           >>>               /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitizer/dataflow/a.dataflow.a934cfd234e75439-cgu.0.rcgu.o:(<&u8 as core::fmt::Debug>::fmt (.dfsan))
           
           rust-lld: error: undefined symbol: <u8 as core::fmt::LowerHex>::fmt (.dfsan)
           >>> referenced by dataflow.a934cfd234e75439-cgu.0
           >>>               /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitizer/dataflow/a.dataflow.a934cfd234e75439-cgu.0.rcgu.o:(<&u8 as core::fmt::Debug>::fmt (.dfsan))
           
           rust-lld: error: undefined symbol: <u8 as core::fmt::UpperHex>::fmt (.dfsan)
           >>> referenced by dataflow.a934cfd234e75439-cgu.0
           >>>               /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitizer/dataflow/a.dataflow.a934cfd234e75439-cgu.0.rcgu.o:(<&u8 as core::fmt::Debug>::fmt (.dfsan))
           
           rust-lld: error: undefined symbol: std::rt::lang_start_internal (.dfsan)
           >>> referenced by dataflow.a934cfd234e75439-cgu.0
           >>>               /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/sanitizer/dataflow/a.dataflow.a934cfd234e75439-cgu.0.rcgu.o:(main)
           collect2: error: ld returned 1 exit status
           

error: aborting due to 1 previous error
------------------------------------------
```

</details>

--- a/src/bootstrap/src/core/builder/cargo.rs
+++ b/src/bootstrap/src/core/builder/cargo.rs
@@ -647,6 +647,8 @@ impl Builder<'_> {
                 // If an explicit setting is given, use that
                 setting
             }
+            // Per compiler-team#938, v0 mangling is used on nightly
+            None if self.config.channel == "dev" || self.config.channel == "nightly" => true,
             None => {
                 if mode == Mode::Std {
                     // The standard library defaults to the legacy scheme

--- a/tests/ui/sanitizer/dataflow-abilist.txt
+++ b/tests/ui/sanitizer/dataflow-abilist.txt
@@ -491,6 +491,8 @@ fun:__dfso_*=discard
 
 # Rust functions.
 fun:_ZN4core*=uninstrumented
+fun:_R*4core*=uninstrumented
 fun:_ZN3std*=uninstrumented
+fun:_R*3std*=uninstrumented
 fun:rust_eh_personality=uninstrumented
 fun:_R*__rustc*=uninstrumented

